utils/matrix/setup_matrix.py

```diff
@@ -0,0 +1,211 @@
+
+    config = ConfigParser()
+    config.read(config_filename)
+
+    matrix_filename = 'conda/logs/matrix.log'
+    if not os.path.exists(matrix_filename):
+        raise OSError(
+            '{} not found.  Try running ./conda/config_compass_env.py to '
+            'generate it.'.format(matrix_filename))
+    with open(matrix_filename, 'r') as f:
+        machine = f.readline().strip()
+        lines = f.readlines()
+        compilers = list()
+        mpis = list()
+        for line in lines:
+            compiler, mpi = line.split(',')
+            compilers.append(compiler.strip())
+            mpis.append(mpi.strip())
+
+    if machine not in all_build_targets:
+        raise ValueError('build targets not known for machine: '
+                         '{}'.format(machine))
+    build_targets = all_build_targets[machine]
+
+    env_name = config.get('matrix', 'env_name')
+
+    openmp = config.get('matrix', 'openmp')
+    openmp = [value.strip().lower() == 'true' for value in
+              openmp.replace(',', '').split(' ')]
+    debug = config.get('matrix', 'debug')
+    debug = [value.strip().lower() == 'true' for value in
+             debug.replace(',', '').split(' ')]
+    other_build_flags = config.get('matrix', 'other_build_flags')
+
+    mpas_path = config.get('matrix', 'mpas_path')
+    mpas_path = os.path.abspath(mpas_path)
+
+    setup_command = config.get('matrix', 'setup_command')
+    work_base = config.get('matrix', 'work_base')
+    work_base = os.path.abspath(work_base)
+    baseline_base = config.get('matrix', 'baseline_base')
+    if baseline_base != '':
+        baseline_base = os.path.abspath(baseline_base)
+
+    for compiler, mpi in zip(compilers, mpis):
+        if (compiler, mpi) not in build_targets:
+            raise ValueError('Unsupported compiler {} and MPI '
+                             '{}'.format(compiler, mpi))
+        target = build_targets[(compiler, mpi)]
+
+        script_name = get_load_script_name(machine, compiler, mpi, env_name)
+        script_name = os.path.abspath(script_name)
+
+        for use_openmp in openmp:
+            for use_debug in debug:
+                suffix = '{}_{}_{}'.format(machine, compiler, mpi)
+                make_command = 'make clean; make {} ' \
+                               '{}'.format(target, other_build_flags)
+                if use_openmp:
+                    make_command = '{} OPENMP=true'.format(make_command)
+                else:
+                    suffix = '{}_noopenmp'.format(suffix)
+                    make_command = '{} OPENMP=false'.format(make_command)
+                if use_debug:
+                    suffix = '{}_debug'.format(suffix)
+                    make_command = '{} DEBUG=true'.format(make_command)
+                else:
+                    make_command = '{} DEBUG=false'.format(make_command)
+                mpas_model = build_mpas(
+                    script_name, mpas_path, make_command, suffix)
```

Comment:
It seems like the `CORE=` in the make command isn't necessary anymore, right? I've always used it, but I bet that's just a habit I've carried over from before the cores got transplanted to E3SM.

Reply:
It isn't needed since the code moved over to E3SM.  The `CORE` is automatically determined based on the directory you're building from as far as I understand it.